sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorPersister.java

```diff
@@ -313,61 +313,76 @@ public void setMany(Map<String, byte[]> unprefixedPathBytesMap) throws Persister
     }
 
     @Override
+    public Map<String, byte[]> getMany(Collection<String> unprefixedPaths) throws PersisterException {
+        if (unprefixedPaths.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        logger.debug("Getting many entries: {}", unprefixedPaths);
+
+        Map<String, byte[]> result = new TreeMap<>();
+        // Unlike with writes, there is not an atomic read operation. Therefore we wing it with a series of plain reads.
```

Comment:
Now that I think of it, we could take a lock in ZK itself to this end? It's not essential now, but I think we should keep it in our back pocket if we see synchronization issues here at some point. Maybe an RW lock with two semaphores?

Reply:
In practice it shouldn't be an issue assuming the PersisterCache is enabled, as in that situation this function wouldn't be getting called anyway (the cache just fetches all data in the tree up-front then serves from that). Added a comment to explain/justify the current winging-of-it.